### PR TITLE
Latest updates ATAC-seq

### PR DESCRIPTION
**Pull request by marinafloresp**
# Description

This pull request will add a script that checks whether all samples have passed through stage 2 of QC. This outputs a progress csv report similar to the one produced in STEP 4.2. Documentation has been updated accordingly. 

In addition, scripts to perform the third stage of the QC have also been added. This will use the counts in peaks obtained in STEP 7.3 to check the cell-type identity of samples. Both main submission script and subscripts have been added. These are still being tested.

## Issue ticket number

This pull request is to address issues: #276  and #273.

## Type of pull request

- [ ] Bug fix
- [X] New feature/enhancement
- [ ] Code refactor
- [X] Documentation update

## Checklist

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have tested my code to check that it is functional
- [ ] I have used linters to check for common sources of errors
- [ ] I have implemented fail safes in my code to account for edge cases
- [ ] I have made the corresponding changes to the documentation

### CODE REVIEW

File: sequencing/ATACSeq/jobSubmission/8_collateStage3QCMatrics.sh [hunk 1]
```diff
@@ -127,7 +127,7 @@ Output directory will be: ${PEAKCOUNTS}
 
 EOF
  
-	Rscript -e "rmarkdown::render(paste0(commandArgs(trailingOnly=TRUE)[1], '/collateCellTypeCheck.Rmd'), output_file=paste0(commandArgs(trailingOnly=TRUE)[2], '/stage3SummaryStats_all.html'))" "${RSCRIPTS_DIR}" "${PEAKCOUNTS}" "${CONFIGR}" "${SETPEAKS}"
+	Rscript -e "rmarkdown::render(paste0(commandArgs(trailingOnly=TRUE)[1], '/collateCellTypeCheck.Rmd'), output_file=paste0(commandArgs(trailingOnly=TRUE)[2], '/stage3SummaryStats_',commandArgs(trailingOnly=TRUE)[4],'.html'))" "${RSCRIPTS_DIR}" "${PEAKCOUNTS}" "${CONFIGR}" "${SETPEAKS}"
```
Comment (sof202) [inline sequencing/ATACSeq/jobSubmission/8_collateStage3QCMatrics.sh:130]: Why are you passing arguments into an `Rscript` call that is being made on the command line? You could just replace this with:
```bash
Rscript -e "rmarkdown::render('${RSCRIPTS_DIR}/collateCellTypeCheck.Rmd'), output_file='${PEAKCOUNTS}/stage3SummaryStats_',commandArgs(trailingOnly=TRUE)[4],'.html'))" "${RSCRIPTS_DIR}" "${PEAKCOUNTS}" "${CONFIGR}" "${SETPEAKS}"
```

(not sure if you need "${RSCRIPTS_DIR}" and "${PEAKCOUNTS}" in the actual `.Rmd` file so I left these in)

File: sequencing/ATACSeq/Rscripts/countsInPeaks.r [hunk 2]
```diff
@@ -29,12 +29,14 @@ args <- commandArgs()
...
-library(TxDb.Hsapiens.UCSC.hg38.knownGene)
-library(org.Hs.eg.db)
+suppressWarnings(suppressPackageStartupMessages({
+  library(Rsubread)
+  library(GenomicRanges)
+  library(ChIPpeakAnno)
+  library(ChIPseeker)
+  library(TxDb.Hsapiens.UCSC.hg38.knownGene)
+  library(org.Hs.eg.db)
+}))
```
Comment (sof202) [inline sequencing/ATACSeq/Rscripts/countsInPeaks.r:39]: @gtw99, A good alternative approach for suppressing unwanted package start up messages.

File: sequencing/ATACSeq/jobSubmission/8_collateStage3QCMatrics.sh [hunk 3]
```diff
@@ -84,7 +84,7 @@ fi
 ## Input in command line what set of peaks to perform analysis on: promoter peaks (prom) or all peaks (all)
 if [[ ! $3 == "PROM" ]] && [[ ! $3 == "ALL" ]];
 then
-  echo "No peak set specified. Please choose either PROM or ALL to select counts in promoter peaks or all peaks, respectively" ; exit 1; }     
+  { echo "No peak set specified. Please choose either PROM or ALL to select counts in promoter peaks or all peaks, respectively" ; exit 1; }
```
Comment (sof202) [inline sequencing/ATACSeq/jobSubmission/8_collateStage3QCMatrics.sh:87]: Good spot, the linter picked this up by I've stopped paying attention to it recently, my bad.

File: sequencing/ATACSeq/subScripts/groupPeaks.sh [hunk 4]
```diff
@@ -45,7 +45,7 @@ echo "Starting peak calling using MACS3 PE on samples that belong to cell fracti
 date -u
 
 cd ${TMPDIR}
-macs3 callpeak -t  ${bamFiles[@]} --outdir ${PEAK_DIR}/MACS/BAMPE/group -n ${GROUP} -f BAMPE -g 2.9e9 -q 5e-2 --keep-dup all --nomodel --broad --broad-cutoff 5e-2
+macs3 callpeak -t  ${bamFiles[@]} --outdir ${PEAK_DIR}/MACS/BAMPE/group -n ${GROUP} -f BAMPE -g 2.9e9 -q 1e-3 --keep-dup all --nomodel --broad --broad-cutoff 1e-3
```
Comment (sof202) [inline sequencing/ATACSeq/subScripts/groupPeaks.sh:48]: Do we want to make the cutoff a user defined variable in the config file or does this not really matter?

Comment (marinafloresp) [reply sequencing/ATACSeq/subScripts/groupPeaks.sh:48]: Yes, it could be a user defined option

**Pull request closed without merging**